### ISSUE TITLE
Importing JSON from Restyaboard

### ISSUE THREAD

**Issue by t52lE4MWEcgCx** [wekan/wekan#3181]
## Import JSON from Restyaboard

[Restyaboard](https://github.com/RestyaPlatform) is another open-source tool, like Wekan. 

On [https://blog.sleeplessbeastie.eu/2018/04/02/how-to-backup-and-restore-restyaboard-boards/
](https://blog.sleeplessbeastie.eu/2018/04/02/how-to-backup-and-restore-restyaboard-boards/
) there is a script to download a JSON-file from Restyaboard (turn TFA off to get it to work). The structure, unfortunately, does not match the Trello-format. 

It would be great if Restyboard-exports can be imported into Wekan to make a switch to Wekan easier. Restya does support Wekan-import. 

Unfortunately, I'm not skilled enough to do this. Sorry.

**Comment by xet7** [comment]
@t52lE4MWEcgCx 

I don't know about anyones skills, but some newbies have just read the code and added features using similar code, [like in this story](https://blog.wekan.team/2018/05/wekan-v1-00-released/). I do encourage anyone to try.

For example, [CSV/TSV import export](https://github.com/wekan/wekan/pulls?q=is%3Apr+is%3Aclosed+csv) is a new one.

There is files here like `import.js`, `trelloCreator.js`, `wekanCreator.js`, `csvCreator.js`:
https://github.com/wekan/wekan/tree/master/models

And menu options are at `sidebar.js` and `sidebar.jade` here:
https://github.com/wekan/wekan/tree/master/client/components/sidebar

Alternatively, it's possible to sponsor me developing this feature with 500€:
https://wekan.team/commercial-support/

**Comment by t52lE4MWEcgCx** [comment]
@xet7 

Thank you for your encouragement. It is much appreciated. And also for hinting at the right files to start with. They look well structured and I might be able to make sense of it. I suppose the result is a variant of trelloCreator.js

My problem starts with setting up the development environment on my Mac. 
Then I've been out of dev for more than 20 years and never used Java or any other OO-languages. That's the lower hurdle, IMHO, compared to setting up the dev-env.

However, I could imagine taking the following approach:
-  create an empty board in Restya - Export it - implement necessary import functionality in Wekan - Test
- create one empty list - export - implement - test
- create one card on that list - export - implement - test
- check for several lists and several cards
- work on card details

**Comment by xet7** [comment]
@t52lE4MWEcgCx 

Developing:
- https://github.com/wekan/wekan/wiki/Mac#build-bundle-from-source-and-develop-wekan
- https://github.com/wekan/wekan/issues/3071#issuecomment-622300835

Please ask immediately as comment in this issue, if you have any questions, so I can point you to correct file where code it.

You could try this:

1) Create script that does have these variables:
- restyaboard_url
- username
- password
- board_id

2) Look at [that bash code](https://blog.sleeplessbeastie.eu/2018/04/02/how-to-backup-and-restore-restyaboard-boards/) and also Wekan import code that is in javascript.

3) Try to import one card from one board, using only javascript, that you can run with `node import.js`, that will save output to file. Continue until you get all data from Restya saved.

4) Create [Import from RestyaBoard form](https://github.com/wekan/wekan/tree/master/client/components/import) where you can input those above 1) options to import from restyaboard.

**Comment by milosz** [comment]
Hello, If you can provide me sample json that can be imported to Wekan then I can create dedicated version for it.
It won't include attachments or comments, just basic card data, although can be extended later.

**Comment by xet7** [comment]
@milosz 

Here is example exported board:
[wekan-export-board.json.txt](https://github.com/wekan/wekan/files/4845868/wekan-export-board.json.txt)

Screenshots:

![example-export-board-1](https://user-images.githubusercontent.com/15545/86013607-54d73880-b9ed-11ea-9077-f737a741373b.png)
 
![example-export-board-2](https://user-images.githubusercontent.com/15545/86013698-6d475300-b9ed-11ea-8fe8-0bbb5c1c6d42.png)

**Comment by milosz** [comment]
Thanks, I will look into this.

**Comment by milosz** [comment]
OK. I have very simple proof of concept.

1. Download Restya JSON  board using shell script from https://blog.sleeplessbeastie.eu/2018/04/02/how-to-backup-and-restore-restyaboard-boards/
[board.json.txt](https://github.com/wekan/wekan/files/4881097/board.json.txt)

2. I have prepared minimal required/working Wekan JSON file. 
[minimal.wekan.json.txt](https://github.com/wekan/wekan/files/4881105/minimal.wekan.json.txt)

3. I have created very simple shell script that provides minimal functionality
[wekan-create-json.sh.txt](https://github.com/wekan/wekan/files/4881114/wekan-create-json.sh.txt)

4. Result can be imported.
[working-import.json.txt](https://github.com/wekan/wekan/files/4881117/working-import.json.txt)

I hope it will get you started. It is quick and dirty solution, Python/Go would be better suited for this task as the idea is quite simple - just parse JSON file to extract lists and cards.

**Comment by milosz** [comment]
I have updated this shell script to include labels an published it online at https://blog.sleeplessbeastie.eu/2020/07/17/how-to-import-restya-board-into-wekan/

**Comment by t52lE4MWEcgCx** [comment]
Thank you, @milosz! Would it be possible for you to also add Checklists? 
Cheers, Ralph

**Comment by milosz** [comment]
It should be possible to add checklists using shell script, but Python would be better suited for this task.
I will try to look into this later this year. Ping me directly as a reminder.

**Comment by t52lE4MWEcgCx** [comment]
Thank you @milosz! I deactivated Restya for now, though. The update process is unclear on docker, the app doesn't work, search is not a basic feature and, though I can export my data based on your scripts, it is not easy to get them elsewhere at the moment. We are back on Trello for now and I will look further into Wekan, though I also run into a data persistence problem when I updated mongodb. Best regards, Ralph
